playground/backend/internal/code_processing/code_processing.go

```diff
@@ -184,7 +188,7 @@ func runCmdWithOutput(cmd *exec.Cmd, stdOutput io.Writer, stdError *bytes.Buffer
 // processStep processes each executor's step with cancel and timeout checks.
 // If finishes by canceling, timeout or error - returns error.
 // If finishes successfully returns nil.
-func processStep(ctx context.Context, pipelineId uuid.UUID, cacheService cache.Cache, cancelChannel, successChannel chan bool, outDataBuffer, errorDataBuffer *bytes.Buffer, errorChannel chan error, errorCaseStatus, successCaseStatus pb.Status) error {
+func processStep(ctx context.Context, pipelineId uuid.UUID, cacheService cache.Cache, cancelChannel, successChannel, stopReadLogsChannel, finishReadLogChannel chan bool, outDataBuffer, errorDataBuffer *bytes.Buffer, errorChannel chan error, errorCaseStatus, successCaseStatus pb.Status) error {
```

Comment:
Incredible arguments number. I don't know how we should reduce it, but this method implements not so good if required ~13 arguments.

Reply:
Totally agree

Reply:
Changed.